src/main.rs

```diff
@@ -549,6 +550,29 @@ async fn consume_messenger_events(
     Ok(())
 }
 
+// CurrentPeers keeps up-to-date with all of the peers we're connected and disconnected to using a map.
+struct CurrentPeers {
+    map: Mutex<HashMap<PublicKey, bool>>,
+}
+
+impl CurrentPeers {
```

Comment:
Can we add the handling in `consume_messenger_events` in this commit rather than combining it with the outgoing message polling commit?